modules/files/files.go

```diff
@@ -80,6 +106,22 @@ func CopyTerragruntFolderToTemp(folderPath string, tempFolderPrefix string) (str
 	return destFolder, nil
 }
 
+// CopyTerragruntFolderToDest creates a copy of the given folder and all its contents in a specified folder with a unique name and the given prefix.
+// Since terragrunt uses tfvars files to specify modules, they are copied to the directory as well.
+// Terraform state files are excluded as well as .terragrunt-cache to avoid overwriting contents.
+func CopyTerragruntFolderToDest(folderPath string, destRootFolder string, tempFolderPrefix string) (string, error) {
+	filter := func(path string) bool {
+		return !PathContainsHiddenFileOrFolder(path) && !PathContainsTerraformState(path)
+	}
+
+	destFolder, err := CopyFolderToDest(folderPath, destRootFolder, tempFolderPrefix, filter)
+	if err != nil {
+		return "", err
+	}
+
+	return destFolder, nil
+}
+
 // CopyFolderToTemp creates a copy of the given folder and all its filtered contents in a temp folder
 // with a unique name and the given prefix.
 func CopyFolderToTemp(folderPath string, tempFolderPrefix string, filter func(path string) bool) (string, error) {
```

Comment:
You should update all the "CopyToTemp" methods to call the equivalent "CopyToDest" methods under the hood, passing `os.TempDir()` ad the `destFolder` so we can reduce code duplication.